datafusion/common/src/config.rs

```diff
@@ -273,6 +273,11 @@ config_namespace! {
         /// memory consumption
         pub max_buffered_batches_per_output_file: usize, default = 2
 
+        /// When scanning file paths, whether to ignore subdirectory files,
+        /// ignored by default (true), when reading a partitioned table,
+        /// `listing_table_ignore_subdirectory` is always equal to false, even if set to true
```

Comment:
I updated the description information, do you agree? @alamb

Reply:
I don't understand what this is trying to say 🤔

Reply:
![image](https://github.com/apache/arrow-datafusion/assets/26200914/4a69a6ff-6ecf-40ca-ae6f-8e55c17f0479)
```sql
## read partition table

CREATE EXTERNAL TABLE csv_with_timestamps (
  name VARCHAR,
  ts TIMESTAMP,
  c_date DATE,
)
STORED AS CSV
PARTITIONED BY (c_date)
LOCATION '../core/tests/data/partitioned_table';

set datafusion.execution.listing_table_ignore_subdirectory = true;

select count(*) from partition_tbale;    ## return 4
```

Reply:
Ah, got it -- thank you -- I will propose a clarification in a follow on PR